src/Appwrite/Certificates/AdapterProvider.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Appwrite\Certificates;
+
+class AdapterProvider
+{
+    private $decisionMaker;
+
+    public function __construct(callable $decisionMaker)
```

Comment:
Not sure why the constructor receive a callback in the class you shared, API/class interface seems very loose this way.